pkg/rbd/rbd_util.go

```diff
@@ -113,8 +114,32 @@ type rbdSnapshot struct {
 	RequestName    string
 }
 
+// ImageFeature represents required image features and value
+type ImageFeature struct {
+	// value is rbd feature value in go-ceph
+	value uint64
+	// needRbdNbd indicates whether this image feature requires an rbd-nbd mounter
+	needRbdNbd bool
+	// required is the image features required for this ImageFeature
+	required []string
+}
+
 var (
-	supportedFeatures = sets.NewString("layering")
+	supportedFeatures = map[string]ImageFeature{
+		"layering": {
+			value:      librbd.RbdFeatureLayering,
```

Comment:
These names will be deprecated with go-ceph v0.3.0 (to be released today) use `FeatureLayering` and similat (without the `Rbd` prefix) instead.

Functions to help mapping the integer values to names, and the other way around have been added too. Have a look at https://github.com/ceph/go-ceph/blob/master/rbd/features.go to see what you can use later today (and update go-ceph in `go.mod`).

Reply:
Thanks for your suggestion. I've seen the implementation and I think It would be great If this PR used these. @Madhu-1 Is it okay to change the go-ceph version in this PR? or do we need another PR to change the go-ceph version?

Reply:
sure , we can do it as part of this PR as a separate commit, but we need to wait till go-ceph makes a new release

Reply:
@nixpanic It would be nice to have a function that checks if a `FeatureSet` has a specific feature. and I think we can confirm that the requirement of the feature is met or not. Is there anything in progress in go-ceph?

```
func (fs *FeatureSet) hasFeature(name string) bool
```

Reply:
No, that is not planned to be included. I would convert the `name` to the bit(s) and check if `fs` has those bits set.

Untested (idea from [the unit tests](https://github.com/ceph/go-ceph/blob/master/rbd/features_test.go#L47)):
```go
fs && FeatureSetFromNames([]string{name}) == FeatureSetFromNames([]string{name})
```

I am not sure how useful such a feature in go-ceph would be. You can file an Issue or PR for the project, and we'll consider/review it. We try to keep go-ceph as simple as possible, and do not try to add additional coating/sugar/utility functions, so we may not want to include this kind of functionality.

Reply:
ok, I agree that go-ceph should be as simple as possible. After go-ceph v0.3.0 is released, I'll fix this PR with the changes. Thanks.

Reply:
Thanks!